### PR TITLE
electrum: ping server before returning client

### PR DESCRIPTION
**Pull request by jp1ac4**
Ping server as an extra connectivity check.

### CODE REVIEW

**Comment by jp1ac4** [comment]
I haven't been able to reproduce the test failures on my local machine. Still investigating.

File: src/bitcoin/electrum/client.rs [hunk 1]
```diff
@@ -57,7 +57,9 @@ impl Client {
     pub fn new(electrum_config: &config::ElectrumConfig) -> Result<Self, Error> {
         // First use a dummy config to check connectivity (no retries, short timeout).
         let dummy_config = Config::builder().retry(0).timeout(Some(3)).build();
+        // Try to ping the server.
         bdk_electrum::electrum_client::Client::from_config(&electrum_config.addr, dummy_config)
+            .and_then(|dummy_client| dummy_client.raw_call("server.ping", []))
```
Comment (darosior) [inline src/bitcoin/electrum/client.rs:62]: What if you just used the `ping` method?
```suggestion
            .and_then(|dummy_client| dummy_client.ping())
```

**Comment by jp1ac4** [comment]
> `lianad` logs for CI failure:

Thanks :pray: 

> Please don't re-push just now, i'm trying to debug on Cirrus. Also, we should address the root cause of the issue, not just throw more retries until it passes CI.

I agree. We probably need to look for something else in the logs when starting electrs to consider it ready to receive the ping request.

**Comment by darosior** [comment]
We could let the more robust solutions as a followup.

**Comment by darosior** [comment]
utACK 537ebb2fb3301e82a297a5c8c4c5043d8d0938e4

**Pull request merged**